### PR TITLE
Reduce tests wall time

### LINKED ISSUE #24518: Make troublesome tests faster or move to heavy
## Reason
The following tests:

```
[300.0s]  TIMEOUT contact/test:mortar_restart.frictional_bouncing_block_action_restart_1 FAILED (TIMEOUT) [previous results: TIMEOUT,FINISHED]
[300.0s]  TIMEOUT tensor_mechanics/test:lagrangian/cartesian/updated/convergence/L.L-large FAILED (TIMEOUT) [previous results: TIMEOUT,FINISHED]
[299.9s]  TIMEOUT tensor_mechanics/test:crystal_plasticity/user_object_based.use_substep_dt FAILED (TIMEOUT) [previous results: TIMEOUT,FINISHED]
[299.9s]  TIMEOUT phase_field/test:GrandPotentialPFM.GrandPotentialStrictMassConservation FAILED (TIMEOUT) [previous results: TIMEOUT,FINISHED]
[299.8s]  TIMEOUT tensor_mechanics/test:crystal_plasticity/stress_update_material_based.use_substep_dt FAILED (TIMEOUT) [previous results: TIMEOUT,FINISHED]
```

...commonly time out when machines get busy. This hampers testing.

See https://civet.inl.gov/job/1554857/ for example.

## Design
Either making them faster (ideal), or move them to heavy/non debug (less ideal).

## Impact
Will hopefully improve the testing process while not removing actual tests.

### PR DESCRIPTION
**Pull request by recuero**
Closes #24518

### CODE REVIEW

File: modules/tensor_mechanics/test/tests/lagrangian/cartesian/updated/convergence/L/tests [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
   issues = '#17472'
   design = 'source/kernels/lagrangian/UpdatedLagrangianStressDivergence.md'
-  [./L-large]
+  [L-large]
     type = CSVDiff
     input = 'large.i'
     csvdiff = 'large_out.csv'
-    requirement = "Quadratic convergence in 3D for PresetBCs on a complicated geometry for small deformations, SMP matches FDP."
+    requirement = "Quadratic convergence in 3D for PresetBCs on a complicated geometry for small "
+                  "deformations, SMP matches FDP."
```
Comment (GiudGiud) [inline modules/tensor_mechanics/test/tests/lagrangian/cartesian/updated/convergence/L/tests:9]:
```suggestion
    requirement = "The system shall achieve quadratic convergence in 3D with present Dirichlet displacement boundary conditions on a complicated geometry for small "
                  "deformations, and the preconditioning for a single matrix shall match finite-difference preconditioning results."
```

File: modules/tensor_mechanics/test/tests/lagrangian/cartesian/updated/convergence/L/tests [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+    # For CI timing requirements
+    method = '!dbg'
+  []
+  [L-small]
     type = CSVDiff
     input = 'small.i'
     csvdiff = 'small_out.csv'
-    requirement = "Quadratic convergence in 3D for PresetBCs on a complicated geometry for large deformations, SMP matches FDP."
+    requirement = "Quadratic convergence in 3D for PresetBCs on a complicated geometry for large "
+                  "deformations, SMP matches FDP."
```
Comment (GiudGiud) [inline modules/tensor_mechanics/test/tests/lagrangian/cartesian/updated/convergence/L/tests:19]:
```suggestion
    requirement = "The system shall achieve quadratic convergence in 3D with present Dirichlet displacement boundary conditions for large deformations, and the preconditioning for a single matrix shall match finite-difference preconditioning results."
```

**Review by GiudGiud — COMMENTED** [review]
requirements format has changed

**Review by GiudGiud — APPROVED** [review]

**Review by loganharbour — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the adjustment!

**Pull request merged**